api/provider/aws/test_aws/task_list_test.go

```diff
@@ -0,0 +1,173 @@
+		},
+		{
+			EntityID:   "tsk_id1",
+			EntityType: "task",
+			Key:        "arn",
+			Value:      "arn:aws:ecs:region:012345678910:task/arn1",
+		},
+		{
+			EntityID:   "tsk_id2",
+			EntityType: "task",
+			Key:        "name",
+			Value:      "tsk_name2",
+		},
+		{
+			EntityID:   "tsk_id2",
+			EntityType: "task",
+			Key:        "environment_id",
+			Value:      "env_id2",
+		},
+		{
+			EntityID:   "tsk_id2",
+			EntityType: "task",
+			Key:        "arn",
+			Value:      "arn:aws:ecs:region:012345678910:task/arn2",
+		},
+		{
+			EntityID:   "env_id1",
+			EntityType: "environment",
+			Key:        "name",
+			Value:      "env_name1",
+		},
+		{
+			EntityID:   "env_id2",
+			EntityType: "environment",
+			Key:        "name",
+			Value:      "env_name2",
+		},
+	}
+
+	for _, tag := range tags {
+		if err := tagStore.Insert(tag); err != nil {
+			t.Fatal(err)
+		}
+	}
+
+	listClusterPagesFN := func(input *ecs.ListClustersInput, fn func(output *ecs.ListClustersOutput, lastPage bool) bool) error {
+		clusterARNs := []*string{
+			aws.String("arn:aws:ecs:region:012345678910:cluster/l0-test-env_id1"),
+			aws.String("arn:aws:ecs:region:012345678910:cluster/l0-test-env_id2"),
+		}
+
+		output := &ecs.ListClustersOutput{}
+		output.SetClusterArns(clusterARNs)
+
+		fn(output, true)
+
+		return nil
+	}
+
+	mockAWS.ECS.EXPECT().
+		ListClustersPages(&ecs.ListClustersInput{}, gomock.Any()).
+		Do(listClusterPagesFN).
+		Return(nil)
+
+	var taskARNs = map[string]map[string][]*string{}
+
+	taskARNs["l0-test-env_id1"] = map[string][]*string{}
+	taskARNs["l0-test-env_id2"] = map[string][]*string{}
+
+	taskARNs["l0-test-env_id1"][ecs.DesiredStatusRunning] = []*string{
+		aws.String("arn:aws:ecs:region:012345678910:task/arn1"),
+	}
+
+	taskARNs["l0-test-env_id1"][ecs.DesiredStatusStopped] = []*string{
+		aws.String("arn:aws:ecs:region:012345678910:task/arn2"),
+	}
+
+	taskARNs["l0-test-env_id2"][ecs.DesiredStatusRunning] = []*string{
+		aws.String("arn:aws:ecs:region:012345678910:task/arn3"),
+	}
+
+	taskARNs["l0-test-env_id2"][ecs.DesiredStatusStopped] = []*string{
+		aws.String("arn:aws:ecs:region:012345678910:task/arn4"),
+	}
+
+	newListTaskPagesFN := func(environmentID string, status string) func(input *ecs.ListTasksInput, fn func(output *ecs.ListTasksOutput, lastPage bool) bool) error {
+		listTaskPagesFN := func(input *ecs.ListTasksInput, fn func(output *ecs.ListTasksOutput, lastPage bool) bool) error {
+
+			output := &ecs.ListTasksOutput{}
+			output.SetTaskArns(taskARNs[environmentID][status])
+			fn(output, true)
+
+			return nil
+		}
+
+		return listTaskPagesFN
+	}
+
+	for _, environmentID := range []string{"l0-test-env_id1", "l0-test-env_id2"} {
+		for _, status := range []string{ecs.DesiredStatusRunning, ecs.DesiredStatusStopped} {
+			listTasksInput := &ecs.ListTasksInput{}
+			listTasksInput.SetCluster(environmentID)
+			listTasksInput.SetDesiredStatus(status)
+			listTasksInput.SetStartedBy("test")
+
+			mockAWS.ECS.EXPECT().
```

Comment:
```go
newListTaskPagesFN := func(i int) func(input *ecs.ListTasksInput, fn func(output *ecs.ListTasksOutput, lastPage bool) bool) error {
    return func(input *ecs.ListTasksInput, fn func(output *ecs.ListTasksOutput, lastPage bool) bool) error {
        var arn string
        switch i {
        case 0:
            arn = "../arn1"
        ...
        default:
            arn = ".../arn4"
        }

        output := &ecs.ListTasksOutput{}
        output.SetTaskArns([]*string{arn})
        fn(output, true)
 
        return nil
    })
}

for i := range  environments {
    for j := range statuses {
        mockAWS.ECS.EXPECT().
            ListTasksPages(listTasksInput, gomock.Any()).
            Do(newListTaskPagesFN(i+j)).
            Return(nil)
    }
}
```

Reply:
you could avoid the `switch` statement since each arn is just a prefix + integer suffix, e.g. 
```
arn := fmt.Sprintf("aws:arn:..../arn%d", i+1)
```